### PR TITLE
Log more helpful error messages for file uploads

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** log frontend file upload errors as an analytics event to start capturing common error conditions.

JIRA: Based on issues in [EDUCATOR-5406](https://openedx.atlassian.net/browse/EDUCATOR-5406)

**What changed?**
- Log an analytics event for file upload errors.

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

**Testing Instructions**

1. Go to an ORA with file uploads enabled
2. Coerce an error on upload, suggest uploading file w/ an unsupported title like `tęšt-fïłê.png`
3. Inspect LMS system logs for an analytics event which should contain the two fields below (plus a lot of context info)

```
  "name": "openassessment.upload_file_error",
  "event": "{\"statusText\": \"TypeError: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': String contains non ISO-8859-1 code point.\"}", 
```

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Comment by muselesscreator** [comment]
I love this in theory, but am a bit worried about logging this level of information to the user.  Is the sort of thing that can be exploited, and otherwise leads to a bad user experience...

**Comment by muselesscreator** [comment]
Any way for the more complex logging message to be network-only, with a UUID logged in the UI to help us search for it in splunk?

**Comment by muselesscreator** [comment]
Can I get a bit more detail on the difference between textStatus and errorThrown (or examples)?

**Comment by nsprenkle** [comment]
> Can I get a bit more detail on the difference between textStatus and errorThrown (or examples)?

Yup! At least for the example error of "You have uploaded a file w/ a bad character set":
- `textStatus: "error"` (existing behavior)
- `errorThrown: "TypeError: Failed to execute 'setRequestHeader' on 'XMLHttpRequest:' String contains non ISO-8859-1 code point."`

I've intentionally set this up to maintain current behavior but to log the detailed error.

**Comment by jansenk** [comment]
How can I best trigger an error locally (I can connect my local devstack to s3 if that makes it easier) ?

**Comment by nsprenkle** [comment]
> How can I best trigger an error locally (I can connect my local devstack to s3 if that makes it easier) ?

One easy error is to try uploading a file with a strange filename: e.g. `tęšt-fïłê.png`.
Other errors you might have to get a bit more creative with.

**Comment by jansenk** [comment]
@nsprenkle When I attempted to do this locally, I saw the error message pop up, but my logged error code was a blank string:
Screen Shot 2020-12-09 at 3 00 05 PM

**Comment by jansenk** [comment]
@muselesscreator could you manually test this and see if you're getting an error? @nsprenkle what backends did you test with? could the error message you're getting be coming from the s3 mocking tool you use?

**Comment by nsprenkle** [comment]
@jansenk, I was testing w/ S3 but didn't expect the behavior to be limited to S3. Will try with other backends.

**Comment by jansenk** [comment]
What browser? Could it be that? I'm using chrome

**Comment by nsprenkle** [comment]
> What browser? Could it be that? I'm using chrome

I'm also using Chrome.

**Comment by nsprenkle** [comment]
@jansenk , this should fix it but please confirm

**Review by jansenk — APPROVED** [review]
Tested manually, looks good 👍  just gotta fix that test

**Review by jnlapierre — APPROVED** [review]
LGTM, code reviewed.
